src/HomePage/components/MovieCard/MovieCard.tsx

```diff
@@ -2,40 +2,48 @@ import React from 'react';
 import Popup from "../Popup/Popup";
 import './MovieCard.css'
 import '../Popup/Popup.css'
-import {useTogglerState} from "./Hooks";
+import {EditMovie} from "../EditMovie/EditMovie";
+import {DeleteMovie} from "../DeleteMovie/DeleteMovie";
 
-interface MovieCardInterface {
+export interface MovieCardInterface {
     title: string,
-    popularity: number
+    popularity: number,
+    onMovieCardClick: ()=> void,
+    onMovieCardEditable: ()=> void,
+    onMovieCardDeletable: ()=> void,
+    isActionsPopupOpen: boolean,
+    isCardEditable: boolean,
+    isCardDeletable: boolean,
 }
 
 export function MovieCard(props: MovieCardInterface) {
-    const [isOpen, toggleIsOpen] = useTogglerState(false);
-    const [isEdited, toggleIsEdited] = useTogglerState(false);
-    const [isDeleted, toggleIsDeleted] = useTogglerState(false);
-
-    const togglePopup = () => {
-        toggleIsOpen();
-    }
-
-    const toggleEdit = () => {
-        toggleIsEdited();
-    }
-
-    const toggleDelete = () => {
-        toggleIsDeleted();
-    }
-
-    return (<div className='card' onClick={togglePopup}>
+    return (<div className='card' onClick={props.onMovieCardClick}>
         <div>{props.title}</div>
 
         <div>{props.popularity}</div>
-        {isOpen && <Popup
+        {props.isActionsPopupOpen && <Popup
+            mode={'small'}
             content={<>
-                <button className="modify-btn" onClick={toggleEdit}>Edit</button> {isEdited}
-                <button className="modify-btn" onClick={toggleDelete}>Delete</button> {isDeleted}
+                <button className="modify-btn" onClick={props.onMovieCardEditable}>Edit</button>
+                <button className="modify-btn" onClick={props.onMovieCardDeletable}>Delete</button>
             </>}
-            handleClose={togglePopup}
+            handleClose={props.onMovieCardClick}
+        />}
+        {props.isCardEditable && <Popup
+            mode={'full'}
+            content={<EditMovie
+                title={'Mocked title'}
+                data={'02-03-2022'}
+                genre={'comedy'}
+                rating={7}
+                runtime={68}
+                url={'http://localhost'}/>}
+            handleClose={props.onMovieCardEditable}
+        />}
```

Comment:
it's not necessary to have all these popups in MovieCard. We can move them to MoviesList level and by doing so, we will reduce the amount of props passed to MovieCard and also make it more readable